docs/interface/json-rpc.md

```diff
@@ -0,0 +1,48 @@
+# JSON-RPC
+
+JSON-RPC is a stateless, light-weight remote procedure call (RPC) protocol.
+Primarily this specification defines several data structures and the rules
+around their processing. It is transport agnostic in that the concepts can
+be used within the same process, over sockets, over http, or in many various
+message passing environments. It uses
+[JSON](http://www.json.org/)
+([RFC 4627](http://www.ietf.org/rfc/rfc4627.txt))
+as data format.
+
+For more details, see the [JSON-RPC 2.0 Specification][json_rpc_specs].
+
+## Server
+
+By default, a JSON-RPC server is started at `127.0.0.1:21338`.
+
+## Protocol
+
+`NewLineCodec`: a final newline (`\n`) is required in order to start processing
+the request.
+
+A message must be a valid utf8 string finished with a newline.
+
+### Methods
+
+See [`json_rpc_methods.rs`][json_rpc_methods] for the implementation
+details.
+
+#### inventory
+
+Publish the received item.
+
+@params: Block
```

Comment:
The input for the `inventory` method is not directly the `block` itself but a discriminated union of `block`, `tx` and others. For example:
```json
{ "type": "block", "data": { ... the block itself...  } }
```

Reply:
I will change it to use an enum:

```rust
/// Inventory element: block, tx, etc
#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
pub enum InvElem {
    #[serde(rename = "block")]
    Block(Block),
    // Other fields...
}
```

Which results in the following JSON:

```js
{"block":{ /* the block itself */ }}
```